### DISCUSSION TITLE
Karma controller

### DISCUSSION THREAD

**Discussion by parviz11** [KonradIT/goprowifihack#267] [Q&A]
Hi,

Would it be possible to mirror Karma controller to iPhone with any of the listed methods?

**Comment by KonradIT** [comment]
Lol I actually bought a Karma drone a year ago and wanted to document the APIs, still have it in my attic.

I think you can use the udp:// ip address :8554 trick with the Karma controller iirc if you used the passenger wifi feature.

**Reply by parviz11** [reply]
Thanks. I will check that.
I was thinking of turning my Karma drone to FPV drone by mirroring controller to a phone and using it with goggles. But this is a project for indefinite future when I get some time from work :D

**Reply by KonradIT** [reply]
The passenger app does not allow for drone control, you can just change camera settings, view media and preview. It creates a tunnel from the RC WiFi server to the camera API. 

I looked at the passenger APIs a year ago when I got the drone: https://twitter.com/konrad_it/status/1338872308371427329

Also @DJISDKUser worked on popping a root shell and more:

- https://twitter.com/d0tslash/status/946239881109491713
- https://twitter.com/d0tslash/status/945355191179796483
- https://twitter.com/d0tslash/status/945375927604858886
- https://twitter.com/d0tslash/status/946839221364576257

Would be interesting to give it a new life. Mine has been sitting unused for a year because I have access to infinitely better drones. No auto landing, or even landing sensors, or any type of sensor whatsoever doom this thing.

**Reply by parviz11** [reply]
I didn’t know Passenger app exists (or existed). Then I went to AppStore and couldn’t find it there. Damn, I should have downloaded it before. Do you happen to know if it is possible to install passenger app manually on iPhone? It has all I need to turn Karma to FPV.

**Reply by KonradIT** [reply]
No idea, could try to use an android phone and the passenger APK: https://apkcombo.com/passenger/com.gopro.passenger/
